samples/host_exerciser/host_exerciser_lpbk.h

```diff
@@ -0,0 +1,253 @@
+     std::cout << "Host Exerciser stride value:" << afu_error1.error << std::endl;
+  }
+
+  void host_exerciser_swtestmsg()
+  {
+    uint64_t swtest_msg;
+    swtest_msg = he_afu_->read64(CSR_SWTEST_MSG);
+
+    std::cout << "Host Exerciser swtest msg:" << swtest_msg << std::endl;
+  }
+
+  inline uint64_t cacheline_aligned_addr(uint64_t num) {
+    return num >> LOG2_CL;
+  }
+
+  int parse_input_options(test_afu *afu,uint32_t *input_cmds)
+  {
+    auto d_afu = dynamic_cast<host_exerciser*>(afu);
+    csr_cfg helpbk_cfg_(0);
+
+    // Host Exerciser Mode
+    if ((d_afu->mode_str_ != "lpbk") &&
+        (d_afu->mode_str_  != "read") &&
+        (d_afu->mode_str_ != "write") &&
+        (d_afu->mode_str_  != "trput")) {
+          std::cerr << "Invalid --mode: " << d_afu->mode_str_ << std::endl;
+          return -1;
+    }
+
+    if (d_afu->mode_str_  == "lpbk") {
+         helpbk_cfg_.cr_mode= HOST_EXEMODE_LPBK1;
+    } else if (d_afu->mode_str_ == "read") {
+         helpbk_cfg_.cr_mode = HOST_EXEMODE_READ;
+    } else if (d_afu->mode_str_ == "write") {
+         helpbk_cfg_.cr_mode = HOST_EXEMODE_WRITE;
+    } else {
+         helpbk_cfg_.cr_mode = HOST_EXEMODE_TRUPT;
+    }
+
+    // Host Exerciser Read
+    if ((d_afu->read_str_ != "rdline_s") &&
+        (d_afu->read_str_ != "rdline_l") &&
+        (d_afu->read_str_ != "mixed")) {
+          std::cerr << "Invalid --read: " << d_afu->read_str_ << std::endl;
+          return -1;
+    }
+
+    if (d_afu->read_str_ == "rdline_s") {
+       helpbk_cfg_.cr_rdsel = HOSTEXE_RDLINE_S;
+    } else if (d_afu->read_str_ == "rdline_l") {
+       helpbk_cfg_.cr_rdsel = HOSTEXE_RDLINE_L;
+    } else {
+       helpbk_cfg_.cr_rdsel = HOSTEXE_RDLINE_MIX;
+    }
+
+    // Host Exerciser Write
+    if ((d_afu->write_str_ != "wrline_m") &&
+      (d_afu->write_str_ != "wrline_l") ) {
+        std::cerr << "Invalid --write: " << d_afu->write_str_ << std::endl;
+        return -1;
+    }
+
+    if (d_afu->write_str_ == "wrline_m") {
+         helpbk_cfg_.cr_wrthru_en = HOSTEXE_WRLINE_M;
+    } else {
+         helpbk_cfg_.cr_wrthru_en = HOSTEXE_WRLINE_L;
+    }
+
+    // Host Exerciser  lpbk delay
+    if (d_afu->delay_)
+         helpbk_cfg_.cr_delay_en = 1;
+
+    // write
+    if ((d_afu->multi_cl_ != 0) &&
+        (d_afu->multi_cl_ != 1) &&
+        (d_afu->multi_cl_ != 3)) {
+         std::cerr << "Invalid --multi_cl_: " << d_afu->multi_cl_ << std::endl;
+          return -1;
+    }
+    helpbk_cfg_.cr_multiCL_len = d_afu->multi_cl_;
+
+    //test rollover or test termination
+    if (d_afu->c_cont_)
+        helpbk_cfg_.c_cont = 1;
+
+    *input_cmds = helpbk_cfg_.value;
+    return 0;
+  }
+
+  virtual int run(test_afu *afu, CLI::App *app)
+  {
+    (void)app;
+
+    auto d_afu = dynamic_cast<host_exerciser*>(afu);
+   he_afu_ = dynamic_cast<host_exerciser*>(afu);
+   uint32_t input_cmds;
+
+	auto ret = parse_input_options(afu,&input_cmds);
```

Comment:
We can possibly use a member variable of type csr_cfg, and pass appropriate member of this in validators for CLI11.
If we do this, we can probably skip this step.

Reply:
Added csr_cfg as member variable